Go language tests need to be serialized, or modified to eliminate TCP port conflicts

When the Go language test suite is run as documented:

```
go test -timeout=20m $(go list github.com/hyperledger/fabric/... | grep -v /vendor/ | grep -v /examples/)
```

Go is running the tests in parallel. This is nice because the test suite runs pretty quickly.

The problem is that this parallelization creates conflicts. Tests that create the same TCP server ports will fail if run in parallel.  Here's what I'm seeing now:

```
--- FAIL: TestExecuteDeployTransaction (0.00s)
    exectransaction_test.go:162: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestExecuteInvokeTransaction (0.00s)
    exectransaction_test.go:293: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestExecuteQuery (0.00s)
    exectransaction_test.go:391: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestExecuteInvokeInvalidTransaction (0.00s)
    exectransaction_test.go:474: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestExecuteInvalidQuery (0.00s)
    exectransaction_test.go:535: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestChaincodeInvokeChaincode (0.00s)
    exectransaction_test.go:606: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestChaincodeQueryChaincode (0.00s)
    exectransaction_test.go:718: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
--- FAIL: TestExecuteDeploySysChaincode (0.00s)
    exectransaction_test.go:861: Error starting peer listener listen tcp 0.0.0.0:40303: bind: address already in use
FAIL
FAIL    github.com/hyperledger/fabric/core/chaincode    0.032s
```

Note that the problem is not with this test, it is the fact that _another_ test that is running concurrently using port 40303.

The big-hammer approach to solve this problem is to serialize the Go tests by adding `-p 1` to the test command:

```
go test -timeout=20m -p 1 $(go list github.com/hyperledger/fabric/... | grep -v /vendor/ | grep -v /examples/)
```

This significantly slows things down, though.

I am about to submit a PR that solves the problem by modifying the port number that is used by the conflicting test. This works, but seems like a band-aid on a bigger problem. If this continue to happen in the future as more tests are added we might need to, for example, document a list of server ports for tests to use and assign each test permanently to a fixed port.

--- a/events/events_test.go
+++ b/events/events_test.go
@@ -148,7 +148,7 @@ func TestMain(m *testing.M) {
 
 	//use a different address than what we usually use for "peer"
 	//we override the peerAddress set in chaincode_support.go
-	peerAddress = "0.0.0.0:40303"
+	peerAddress = "0.0.0.0:50303"
 
 	lis, err := net.Listen("tcp", peerAddress)
 	if err != nil {

